src/notebook/plugin.ts

```diff
@@ -64,67 +64,56 @@ const DIRTY_CLASS = 'jp-mod-dirty';
 
 
 /**
- * Register the plugin contributions.
- *
- * @param container - The di container for type registration.
- *
- * #### Notes
- * This is called automatically when the plugin is loaded.
+ * The notebook file handler provider.
  */
 export
-function resolve(container: Container): Promise<AbstractFileHandler> {
-  return container.resolve({
-    requires: [IServicesProvider, IDocumentManager, ICommandRegistry, ICommandPalette],
-    create: (services: IServicesProvider, manager: IDocumentManager,
-             registry: ICommandRegistry,
-             palette: ICommandPalette) => {
-      let handler = new NotebookFileHandler(
-        services.contentsManager,
-        services.notebookSessionManager
-      );
-      manager.register(handler);
-      registry.add([{
-        id: executeCellCommandId,
-        handler: () => handler.executeSelectedCell()
-      }, {
-        id: renderCellCommandId,
-        handler: () => handler.renderSelectedCell()
-      }, {
-        id: selectNextCellCommandId,
-        handler: () => handler.selectNextCell()
-      }, {
-        id: selectPreviousCellCommandId,
-        handler: () => handler.selectPreviousCell()
-      }]);
-      palette.add([{
-        id: executeCellCommandId,
-        category: 'Notebook Operations',
-        args: void 0,
-        text: 'Execute current cell',
-        caption: 'Execute the current cell'
-      }, {
-        id: renderCellCommandId,
-        category: 'Notebook Operations',
-        args: void 0,
-        text: 'Render current markdown cell',
-        caption: 'Render the current markdown cell'
-      }, {
-        id: selectNextCellCommandId,
-        category: 'Notebook Operations',
-        args: void 0,
-        text: 'Select next cell',
-        caption: 'Select next cell'
-      }, {
-        id: selectPreviousCellCommandId,
-        category: 'Notebook Operations',
-        args: void 0,
-        text: 'Select previous cell',
-        caption: 'Select previous cell'
-      }]);
-      return handler;
-    }
-  });
-}
+const fileHandlerProvider = {
```

Comment:
We should talk about whether it's appropriate for a service to define commands. I think it may be better for services to expose rich APIs, and let extensions wrap them up as commands as they see fit. Not 100% convinced about this, but I'm leaning that way.